test/data/azure/main.tf

```diff
@@ -0,0 +1,171 @@
+
+}
+
+# Create subnet
+resource "azurerm_subnet" "testterraformsubnet" {
+    name                 = "testSubnet"
+    resource_group_name  = data.azurerm_resource_group.testResourceGroup.name
+    virtual_network_name = azurerm_virtual_network.testterraformnetwork.name
+    address_prefixes       = ["10.0.1.0/24"]
+}
+
+# Create public IPs
+resource "azurerm_public_ip" "testterraformpublicip" {
+    name                         = "testPublicIP"
+    location                     = data.azurerm_resource_group.testResourceGroup.location
+    resource_group_name          = data.azurerm_resource_group.testResourceGroup.name
+    allocation_method            = "Dynamic"
+
+}
+
+# Create Network Security Group and rule
+resource "azurerm_network_security_group" "testterraformnsg" {
+    name                = "testNetworkSecurityGroup"
+    location            = data.azurerm_resource_group.testResourceGroup.location
+    resource_group_name = data.azurerm_resource_group.testResourceGroup.name
+
+    security_rule {
+        name                       = "SSH"
+        priority                   = 1001
+        direction                  = "Inbound"
+        access                     = "Allow"
+        protocol                   = "Tcp"
+        source_port_range          = "*"
+        destination_port_range     = "22"
+        source_address_prefix      = "*"
+        destination_address_prefix = "*"
+    }
+
+}
+
+# Create network interface
+resource "azurerm_network_interface" "testterraformnic" {
+    name                      = "testNIC"
+    location                  = data.azurerm_resource_group.testResourceGroup.location
+    resource_group_name       = data.azurerm_resource_group.testResourceGroup.name
+
+    ip_configuration {
+        name                          = "testNicConfiguration"
+        subnet_id                     = azurerm_subnet.testterraformsubnet.id
+        private_ip_address_allocation = "Dynamic"
+        public_ip_address_id          = azurerm_public_ip.testterraformpublicip.id
+    }
+
+}
+
+# Connect the security group to the network interface
+resource "azurerm_network_interface_security_group_association" "test" {
+    network_interface_id      = azurerm_network_interface.testterraformnic.id
+    network_security_group_id = azurerm_network_security_group.testterraformnsg.id
+}
+
+# Create (and display) an SSH key
+resource "tls_private_key" "test_ssh" {
+  algorithm = "RSA"
+  rsa_bits = 4096
+}
+output "tls_private_key" {
+    value = tls_private_key.test_ssh.private_key_pem
+    sensitive = true
+}
+
+# Create virtual machine
+resource "azurerm_linux_virtual_machine" "testterraformvm" {
+    name                  = "testVM"
+    location              = data.azurerm_resource_group.testResourceGroup.location
+    resource_group_name   = data.azurerm_resource_group.testResourceGroup.name
+    network_interface_ids = [azurerm_network_interface.testterraformnic.id]
+    size                  = "Standard_DS1_v2"
```

Comment:
Can we try using something cheaper? B1S might work for us.

Reply:
lorax-composer used Standard_B2s, while `test/data/azure/deployment-template.json` uses Standard_B1s

Reply:
There are no `B-series` machines available as `Spot` instances. The cheapest ones are `D-series` and since it's up a minute or two tops, I guess we'll be fine.

Reply:
Looking at the Azure calculator if we take the B1S as the base price then B2S is 4x more expensive while DS1-v2 is 7x more expensive. 

Azure says that is supports "per minute" billing unlike AWS which bills you per "started hour" which is nice.  OTOH my experience with spot instances in AWS is that while most of the time they are indeed very cheap during rush periods (e.g. major holidays) they become more expensive even than their regular price.

I don't have metrics about the hours we consume but it looks to me that a `Standard_B2s` instance is a good compromise.

@ondrejbudai what do you think ?

Reply:
I looked at this again and found out there's `A-series` and that seems to be actually cheaper when used as a `Spot` instance. Also the eviction rate seems a lot lower and they suggest this series to be used for testing, so that might be another option to consider. I don't know what's the difference if we run a Spot instance for 2 minutes or regular instance for 2 minutes though...
![azure_prices](https://user-images.githubusercontent.com/55534579/116886837-dc9f5c00-ac29-11eb-9901-292a4ace8d20.png)

Reply:
@yuxisun1217 do you have any experience insights here? We're looking for the cheapest possible instance to use for testing and preferably use spot instances to go even cheaper. Do you have any practical experience on this subject ?

Reply:
we agreed on our weekly meeting that we will use a B1S non-spot instance for now and we can iterate from there.

Reply:
`B1s` seems to be working fine, all stages passed.